assets/js/modules/adsense/common/use-snippet-switch.test.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * AdSense Use Snippet Switch component tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetchMock from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import UseSnippetSwitch from './use-snippet-switch';
+import { fireEvent, muteConsole, render } from '../../../../../tests/js/test-utils';
+import { STORE_NAME } from '../datastore/constants';
+
+// Mock apiFetch so we know if it's called.
+jest.mock( '@wordpress/api-fetch' );
+apiFetchMock.mockImplementation( ( ...args ) => {
+	// eslint-disable-next-line no-console
+	console.warn( 'apiFetch', ...args );
+} );
+
+const setupRegistry = ( registry ) => {
+	registry.dispatch( STORE_NAME ).receiveSettings( {
+		useSnippet: false,
+	} );
+};
+
+describe( 'UseSnippetSwitch', () => {
+	afterEach( () => apiFetchMock.mockClear() );
+	afterAll( () => jest.restoreAllMocks() );
```

Comment:
Why restore all mocks rather than the particular `apiFetchMock` made here? This seems like it could affect other tests is all.